core/src/alpenglow_consensus/voting_loop.rs

```diff
@@ -938,86 +984,112 @@ impl VotingLoop {
-        if !bank.is_startup_verification_complete() {
-            info!("startup verification incomplete, so unable to vote");
-            return GenerateVoteTxResult::Failed;
-        }
-
-        if authorized_voter_keypairs.is_empty() {
-            return GenerateVoteTxResult::NonVoting;
-        }
-        if let Some(slot) = context.wait_to_vote_slot {
-            if vote.slot() < slot {
+        let authorized_voter_keypair;
+        let bls_pubkey_in_vote_account;
+        {
+            let authorized_voter_keypairs = context.authorized_voter_keypairs.read().unwrap();
+            if !bank.is_startup_verification_complete() {
+                info!("startup verification incomplete, so unable to vote");
                 return GenerateVoteTxResult::Failed;
             }
-        }
-        let vote_account = match bank.get_vote_account(&context.vote_account_pubkey) {
-            None => {
+            if authorized_voter_keypairs.is_empty() {
+                return GenerateVoteTxResult::NonVoting;
+            }
+            if let Some(slot) = context.wait_to_vote_slot {
+                if vote.slot() < slot {
+                    return GenerateVoteTxResult::Failed;
+                }
+            }
+            let Some(vote_account) = bank.get_vote_account(&context.vote_account_pubkey) else {
                 warn!("Vote account {vote_account_pubkey} does not exist.  Unable to vote");
                 return GenerateVoteTxResult::Failed;
-            }
-            Some(vote_account) => vote_account,
-        };
-        let vote_state = match vote_account.alpenglow_vote_state() {
-            None => {
+            };
+            let Some(vote_state) = vote_account.alpenglow_vote_state() else {
                 warn!(
                     "Vote account {vote_account_pubkey} does not have an Alpenglow vote state.  Unable to vote",
                 );
                 return GenerateVoteTxResult::Failed;
+            };
+            if *vote_state.node_pubkey() != context.identity_keypair.pubkey() {
+                info!(
+                    "Vote account node_pubkey mismatch: {} (expected: {}).  Unable to vote",
+                    vote_state.node_pubkey(),
+                    context.identity_keypair.pubkey()
+                );
+                return GenerateVoteTxResult::HotSpare;
             }
-            Some(vote_state) => vote_state,
-        };
-        if *vote_state.node_pubkey() != context.identity_keypair.pubkey() {
-            info!(
-                "Vote account node_pubkey mismatch: {} (expected: {}).  Unable to vote",
-                vote_state.node_pubkey(),
-                context.identity_keypair.pubkey()
-            );
-            return GenerateVoteTxResult::HotSpare;
-        }
+            bls_pubkey_in_vote_account = match vote_account.bls_pubkey() {
+                None => {
+                    warn!(
```

Comment:
Prefer to panic here, the operator must migrate their vote account. Can add instructions once we figure out Vote State v4